src/nvim/tui/tui.c

```diff
@@ -527,7 +529,13 @@ static void tui_flush(UI *ui)
   }
 
   unibi_goto(ui, data->row, data->col);
+
+  if (!data->busy) {
+    unibi_out(ui, unibi_cursor_normal);
+  }
+
   flush_buf(ui);
+  unibi_out(ui, unibi_cursor_invisible);
```

Comment:
This is called so the cursor will  be disabled during updates that follow the flush, but it would be better to only call this if there was an update that required moving the cursor to draw on different parts of the screen. 

I will think of a way to optimize this